src/controllers/accountController.js

```diff
@@ -0,0 +1,104 @@
+const stripe = require("stripe")(process.env.STRIPE_API_KEY);
+const request = require("request");
+const passport = require("passport");
+const userQueries = require("../db/queries.users");
+const wikiQueries = require("../db/queries.wikis");
+
+module.exports = {
+    upgradeForm(req, res, next){
+        if(req.user && req.user.role == 0){
+            res.render("account/upgrade");
+        } else if(!req.user) {
+            req.flash("notice", "You must be signed in to do that.");
+            res.redirect("/users/sign_in");
+        } else if (req.user && req.user.role == 1){
+            res.render("account/upgrade");
+            req.flash("notice", "You are already a premium member. Do you wish to downgrade?");
+        }
+    },
+    upgrade(req, res, next){
+        if(req.body.stripeToken){
+            let options = {
+                amount: 1500,
+                source: req.body.stripeToken,
+                currency: 'usd',
+                description: "Upgrade Account",
+                name: req.body.name,
+            }
+            stripe.charges.create(options, (err, charge) => {
+                if(charge){
+                    userQueries.upgradeUser(req, (err, user) => {
+                        if(err){
+                            req.flash("error", err);
+                            console.log(err);
+                            res.redirect("/account/upgrade");
+                        } else {
+                            req.flash("notice", "Your account has been upgraded!");
+                            res.redirect("/");
+                        }
+                    });
+                } else {
+                    console.log(err);
+                    console.log("REDIRECTING");
+                    res.redirect("/account/upgrade");
```

Comment:
irl youd probably want to redirect them to an error page. You generally want to avoid silent fails, where something fails but failure is never conveyed to the user in any way